File: CS39000-CP2/week2/Ants.java
import java.util.*;
import java.lang.*;

public class Ants {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int t = scan.nextInt();
        for (int i = 0; i < t; i++) {
            int l = scan.nextInt();
            int n = scan.nextInt();

            int ant;
            int shortest = 0;
            int longest = 0;
            for (int j = 0; j < n; j++) {
                ant = scan.nextInt();
                shortest = Math.max(shortest, Math.min(ant, l - ant));
                longest = Math.max(longest, Math.max(ant, l - ant));
            }
            
            System.out.println(shortest + " " + longest);
        }
    }
}

File: CS39000-CP2/week2/Ants.cpp
#include <cstdio>
#include <cstring>
#include <iostream>
#include <stdio.h>

using namespace std;

int min(int a, int b) {
    return a <= b ? a : b;
}
int max(int a, int b) {
    return a >= b ? a : b;
}

int main() {
    int t, l, n, ant, shortest, longest;
    // cin >> t;
    scanf("%i", &t);

    for (int i = 0; i < t; i++) {
        // cin >> l >> n;
        scanf("%i %i", &l, &n);
        shortest = 0;
        longest = 0;

        for (int j = 0; j < n; j++) {
            // cin >> ant;
            scanf("%i", &ant);
            shortest = max(shortest, min(ant, l - ant));
            longest = max(longest, max(ant, l - ant));
        }

        cout<< shortest << " " << longest << endl;
    }
            
    return 0;
}